SampleProject/VOKCoreDataManagerTests/VOKThreadingTests.m

```diff
@@ -0,0 +1,37 @@
+//
+//  VOKThreadingTests.m
+//  Vokoder Sample Project
+//
+//  Created by Brock Boland on 2/26/16.
+//  Copyright © 2016 Vokal. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import <VOKCoreDataManager.h>
+
+@interface VOKThreadingTests : XCTestCase
+
+@end
+
+@implementation VOKThreadingTests
+
+- (void)testSetupOnBackgroundThreadFails
+{
+    XCTestExpectation *willFinish = [self expectationWithDescription:@"Background thread setup will finish"];
+
+    // Setup on the main thread should work
+    [[VOKCoreDataManager sharedInstance] resetCoreData];
+    XCTAssertNoThrow([[VOKCoreDataManager sharedInstance] setResource:@"VOKCoreDataModel" database:nil]);
+
+    NSOperationQueue *backgroundQueue = [[NSOperationQueue alloc] init];
+    backgroundQueue.qualityOfService = NSOperationQualityOfServiceBackground;
+    [backgroundQueue addOperationWithBlock:^{
+        [[VOKCoreDataManager sharedInstance] resetCoreData];
+        // Setup on a background thread should fail
```

Comment:
If you can create the MOC without problem on a background thread, I'd argue that the assertion you added isn't needed.  If you use the main context directly on the main thread and use the `performBlock` methods when on a different thread things should work. If you're worried about problems surfacing later, I don't know if the main queue assertion will help. Do you have an example of problems or just an abundance of caution?